ui/pages/send/send.js

```diff
@@ -95,13 +96,23 @@ export default function SendTransactionScreen() {
         userInput={userInput}
         className="send__to-row"
         onChange={(address) => dispatch(updateRecipientUserInput(address))}
-        onValidAddressTyped={(address) =>
-          dispatch(updateRecipient({ address, nickname: '' }))
-        }
+        onValidAddressTyped={(address) => {
+          dispatch(
+            addHistoryEntry(`sendFlow - Valid address typed ${address}`),
+          );
+          dispatch(updateRecipient({ address, nickname: '' }));
+        }}
         internalSearch={isUsingMyAccountsForRecipientSearch}
         selectedAddress={recipient.address}
         selectedName={recipient.nickname}
-        onPaste={(text) => updateRecipient({ address: text, nickname: '' })}
+        onPaste={(text) => {
+          dispatch(
+            addHistoryEntry(
+              `sendFlow - User pasted ${text} into address field`,
+            ),
+          );
+          updateRecipient({ address: text, nickname: '' });
```

Comment:
so.........uh, i'm not sure how this was working before because it wasn't being dispatched. The likely situation is that the onValidAddressTyped handler is also fired on a paste...